File: src/main/java/com/github/avarabyeu/jashing/events/ProgressBarEvent.java
package com.github.avarabyeu.jashing.events;

import com.github.avarabyeu.jashing.core.JashingEvent;
import com.google.gson.annotations.SerializedName;

import java.util.Collection;

/**
 * @author Andrei Varabyeu
 */
public class ProgressBarEvent extends JashingEvent {

    @SerializedName("progress_items")
    private Collection<Item> items;

    public ProgressBarEvent(String title, Collection<Item> items) {
        setTitle(title);
        this.items = items;
    }

    public ProgressBarEvent(Collection<Item> items) {
        this(null, items);
    }


    public Collection<Item> getItems() {
        return items;
    }

    public void setItems(Collection<Item> items) {
        this.items = items;
    }

    public static class Item {
        private String name;
        private byte progress;

        public Item(String name, byte progress) {
            this.name = name;
            this.progress = progress;
        }

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }

        public byte getProgress() {
            return progress;
        }

        public void setProgress(byte progress) {
            this.progress = progress;
        }
    }

}

File: jashing/src/main/kotlin/com/github/avarabyeu/jashing/events/ProgressBarEvent.kt
package com.github.avarabyeu.jashing.events

import com.github.avarabyeu.jashing.core.JashingEvent
import com.google.gson.annotations.SerializedName

/**
 * @author Andrei Varabyeu
 */
class ProgressBarEvent(title: String?,
                       @SerializedName("progress_items")
                       var items: Collection<ProgressBarEvent.Item>?) : JashingEvent() {
    init {
        this.title = title
    }

    constructor(items: Collection<Item>) : this(null, items) {
    }

    class Item(var name: String?, var progress: Byte)

}
